src/datachain/query/dataset.py

```diff
@@ -601,13 +605,10 @@ def apply(
         if self.partition_by is not None:
             partition_tbl = self.create_partitions_table(query)
             temp_tables.append(partition_tbl.name)
-
-            subq = query.subquery()
-            query = (
-                sqlalchemy.select(*subq.c)
-                .outerjoin(partition_tbl, partition_tbl.c.sys__id == subq.c.sys__id)
-                .add_columns(*partition_columns())
-            )
+            query = query.outerjoin(
+                partition_tbl,
+                partition_tbl.c.sys__id == query.selected_columns.sys__id,
+            ).add_columns(*partition_columns())
```

Comment:
Another major important change is here.
Instead of using subquery, use query itself.
This will allow us to later reset offset, limit and order by, otherwise it will be tricky (to traverse through the whole query with all subqueries any modify each one if needed is not very stable and also may leads to random bugs in future).